src/target_set.c

```diff
@@ -28,6 +28,28 @@
 
 #include "common.h"
 #include "filter_chain.h"
+#include "meta_data.h"
+#include "utils_subst.h"
+
+struct ts_key_list_s
```

Comment:
That's fine, it was just a suggestion.

Utility functions for general linked lists are in `src/daemon/utils_llist.h`, if you want to take a look at that. Since you only use *append* and iterate over the list, the benefits of switching to that are probably few.